ast/ast.go

```diff
@@ -184,18 +186,25 @@ type SubGraph struct {
 }
 
 func NewSubGraph(id, l Attrib) (*SubGraph, error) {
-	g := &SubGraph{ID: ID(fmt.Sprintf("anon%d", r.Int63()))}
-	if id != nil {
-		if len(id.(ID)) > 0 {
-			g.ID = id.(ID)
-		}
+	g := &SubGraph{}
+	if id == nil {
+		g.ID = ID(fmt.Sprintf("anon%d", randInt63()))
+	} else if len(id.(ID)) > 0 {
```

Comment:
Actually here is a "bug"
if id != nil, but len(id.(ID) == 0, then g.ID stays empty

Reply:
i ran out of sane identifiers there

Reply:
How about changing `id` to `maybeId`